File: api/src/Domain/ProductCatalog/Entities/Pad.cs
using System;
using System.Collections.Generic;
using DetailingArsenal.Domain.Shared;

namespace DetailingArsenal.Domain.ProductCatalog {
    public class Pad : Entity<Pad> {
        public string Name { get; set; }
        public List<PadCategory> Category { get; set; }
        public PadMaterial? Material { get; set; }
        public PadTexture? Texture { get; set; }
        public PadColor? Color { get; set; }
        public bool? HasCenterHole { get; set; }
        public ProcessedImage? Image { get; set; }
        public List<PadOption> Options { get; set; }

        public Pad(string name, PadCategory category, PadMaterial? material, PadTexture? texture, PadColor? color, bool? hasCenterHole, ProcessedImage? image = null, List<PadOption>? options = null) {
            Id = Guid.NewGuid();
            Name = name;
            Category = new(new[] { category });
            Material = material;
            Texture = texture;
            Color = color;
            HasCenterHole = hasCenterHole;
            Image = image;
            Options = options ?? new();
        }

        public Pad(string name, List<PadCategory> category, PadMaterial? material, PadTexture? texture, PadColor? color, bool? hasCenterHole, ProcessedImage? image = null, List<PadOption>? options = null) {
            Id = Guid.NewGuid();
            Name = name;
            Category = category;
            Material = material;
            Texture = texture;
            Color = color;
            HasCenterHole = hasCenterHole;
            Image = image;
            Options = options ?? new();
        }

        public Pad(Guid id, string name, List<PadCategory> category, PadMaterial? material, PadTexture? texture, PadColor? color, bool? hasCenterHole, ProcessedImage? image = null, List<PadOption>? options = null) {
            Id = id;
            Name = name;
            Category = category;
            Material = material;
            Texture = texture;
            Color = color;
            HasCenterHole = hasCenterHole;
            Image = image;
            Options = options ?? new();
        }
    }
}

File: app/src/api/admin/data-transfer-objects/pad.ts
import { PadFinish } from '../../product-catalog/data-transfer-objects/pad-finish';
import { PadCut } from '../../product-catalog/data-transfer-objects/pad-cut';
import { Rating } from '../../product-catalog/data-transfer-objects/rating';
import { Image } from '../../shared/data-transfer-objects/image';
import { PadOption } from './pad-option';
import { PadTexture } from '../../shared/data-transfer-objects/pad-texture';
import { PadColor } from '../../shared/data-transfer-objects/pad-color';
import { PadSeries } from '..';
import { PadCategory, PadMaterial } from '@/api/shared';

export class Pad {
    get imageUrl() {
        return this.image != null ? `${process.env.VUE_APP_API_DOMAIN}/images/${this.image}` : null;
    }

    get thumbnailUrl() {
        return this.image != null ? `${process.env.VUE_APP_API_DOMAIN}/images/${this.image}/thumbnail` : null;
    }

    constructor(
        public id: string,
        public name: string,
        public category: PadCategory[],
        public material: PadMaterial | null,
        public texture: PadTexture | null,
        public color: PadColor | null,
        public hasCenterHole: boolean | null,
        public cut: number | null, // Comes from review
        public finish: number | null, // Comes from review
        public rating: Rating,
        public image: string | null = null,
        public options: PadOption[] = []
    ) {}
}
